anyway/views/news_flash/api.py

```diff
@@ -1,46 +1,70 @@
+# pylint: disable=no-name-in-module
+# pylint: disable=no-self-argument
+
 import datetime
 import json
 import logging
-from typing import List
+from typing import List, Optional
 
-from flask import request, Response
+from flask import request, Response, make_response, jsonify
 from sqlalchemy import and_, not_
 
 from anyway.app_and_db import db
 from anyway.backend_constants import BE_CONST
 from anyway.models import NewsFlash
 from anyway.infographics_utils import is_news_flash_resolution_supported
 
+from pydantic import BaseModel, ValidationError, validator
+
 DEFAULT_OFFSET_REQ_PARAMETER = 0
 DEFAULT_LIMIT_REQ_PARAMETER = 100
 
 
+class NewsFlashQuery(BaseModel):
+
+    id: Optional[int]
+    road_number: Optional[int]
+    offset: Optional[int] = DEFAULT_OFFSET_REQ_PARAMETER
+    limit: Optional[int] = DEFAULT_LIMIT_REQ_PARAMETER
+    interurban_only: Optional[bool]
+    road_segment_only: Optional[bool]
+    source: Optional[str]
+    # Must set default value in order to be accessed from "check_missing_date" validator when no value provided
+    # from the request
+    start_date: Optional[datetime.datetime] = None
+    end_date: Optional[datetime.datetime] = None
+
+    @validator("end_date", always=True)
+    def check_missing_date(cls, v, values):
+        if bool(v) != bool(values["start_date"]):
+            raise ValueError("Must provide both start and end date")
+        return v
+
+    @validator("source")
+    def check_source_exist(cls, v):
+        valid_sources = [
+            str(source_name[0])
+            for source_name in db.session.query(NewsFlash.source).distinct().all()
```

Comment:
There should be a much faster way to get the list of sources. As there is specific code for each, it make sense that we can have a hard-coded list of the sources.